spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -168,19 +172,105 @@ public void setExtendedBindingProperties(RabbitExtendedBindingProperties extende
 	}
 
 	@Override
-	public void onInit() {
+	public void onInit() throws Exception {
+		super.onInit();
+
+		CachingConnectionFactory producerConnectionFactory = createProducerConnectionFactory(this.rabbitProperties);
+		producerConnectionFactory.setApplicationContext(getApplicationContext());
+		getApplicationContext().addApplicationListener(producerConnectionFactory);
+		producerConnectionFactory.afterPropertiesSet();
+
+		this.producerConnectionFactory = producerConnectionFactory;
+
 		if (this.clustered) {
 			String[] addresses = StringUtils.commaDelimitedListToStringArray(this.rabbitProperties.getAddresses());
+
 			Assert.state(addresses.length == this.adminAddresses.length
-							&& addresses.length == this.nodes.length,
+					&& addresses.length == this.nodes.length,
 					"'addresses', 'adminAddresses', and 'nodes' properties must have equal length");
+
 			this.connectionFactory = new LocalizedQueueConnectionFactory(this.connectionFactory, addresses,
 					this.adminAddresses, this.nodes, this.rabbitProperties.getVirtualHost(),
 					this.rabbitProperties.getUsername(), this.rabbitProperties.getPassword(),
 					this.rabbitProperties.getSsl().isEnabled(), this.rabbitProperties.getSsl().getKeyStore(),
 					this.rabbitProperties.getSsl().getTrustStore(),
 					this.rabbitProperties.getSsl().getKeyStorePassword(),
 					this.rabbitProperties.getSsl().getTrustStorePassword());
+
+			this.producerConnectionFactory = new LocalizedQueueConnectionFactory(this.producerConnectionFactory,
+					addresses, this.adminAddresses, this.nodes, this.rabbitProperties.getVirtualHost(),
+					this.rabbitProperties.getUsername(), this.rabbitProperties.getPassword(),
```

Comment:
We don't need this on the producer side; we only need queue affinity for consumers.